homeassistant/components/mqtt/__init__.py

```diff
@@ -957,7 +958,9 @@ def _mqtt_on_disconnect(self, _mqttc, _userdata, result_code: int) -> None:
 
     async def _wait_for_mid(self, mid):
         """Wait for ACK from broker."""
-        self._pending_operations[mid] = asyncio.Event()
+        async with self._pending_operations_lock:
```

Comment:
We don't need a lock here.

Reply:
Right, both `_wait_for_mid` and `_mqtt_handle_mid` will be called from the event loop so there is no need for any lock when accessing self._pending_operations.